### ISSUE TITLE
[BUG] `FeatureUnion` raises error with transformers with primitive output

### ISSUE THREAD

**Issue by fspinna** [sktime/sktime#6077]
**Describe the bug**
Feature Union raises error with transformers with primitive output, i.e., when the output is not a series or a panel sktime raises an exception.

**To Reproduce**
```python
from sktime.datasets import load_gunpoint
from sktime.transformations.panel.rocket import Rocket
X_train, y_train = load_gunpoint(split="train", return_X_y=True)
union = Rocket() + Rocket()
X_train_transformed = union.fit_transform(X_train)
```

raises: 
```
No valid mtype could be identified for object of type <class 'pandas.core.frame.DataFrame'>. Errors returned are as follows, in format [mtype]: [error message] 
df-list: obj must be list of pd.DataFrame, found <class 'pandas.core.frame.DataFrame'>
numpy3D: obj must be a numpy.ndarray, found <class 'pandas.core.frame.DataFrame'>
pd-multiindex: obj must have a MultiIndex, found <class 'pandas.core.indexes.range.RangeIndex'>
nested_univ: obj All columns must be object, found <class 'pandas.core.frame.DataFrame'>
```

**Expected behavior**
I think the expected behavior should be to simply concatenate the primitives on axis=1. With `set_config(**{"output_conversion": "off"})` the transformation seems to work fine.

**Additional context**
Maybe this has something to do with this to-do in the init:
```python
# todo: check for transform-input, transform-output
#   for now, we assume it's always Series/Series or Series/Panel
#   but no error is currently raised
```

**Versions**
System:
    python: 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:42:20)  [Clang 14.0.6 ]
   machine: macOS-13.2.1-x86_64-i386-64bit

Python dependencies:
          pip: 23.2.1
       sktime: 0.27.0
      sklearn: 1.3.0
       skbase: 0.7.5
        numpy: 1.24.0
        scipy: 1.11.1
       pandas: 2.0.3
   matplotlib: 3.7.2
       joblib: 1.3.1
        numba: 0.57.1
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

**Comment by fkiraly** [comment]
this might relate to recent changes introduced in https://github.com/sktime/sktime/pull/5351 - or not. Either way, should not be too difficult to fix by going through the output conversion cases.

**Comment by fkiraly** [comment]
seems like the `output_scitype` variable is `"Series"` here, incorrectly inferred

**Comment by fkiraly** [comment]
no, not actually related to #5351.

The diagnosis of @fspinna was accurate, it was the missing logic for setting the output tag correctly.
